upa/contracts/UpaVerifier.sol

```diff
@@ -618,6 +622,161 @@ contract UpaVerifier is
+    /// `submissionProofs` - Merkle proofs, each showing that each interval of
+    /// proofIds belongs to an on-chain submission
+    /// `offChainSubmissionMarkers` - encodes a bool[256] where a `1` marks
+    /// each proofId that is at the end of an off-chain submission.
+    /// `duplicateSubmissionIndices` - packed uint8s, reading the lowest-order
+    /// byte first, indicating the index of the duplicated submission.
+    function verifyMixedAggregatedProof(
+        bytes calldata proof,
+        bytes32[] calldata proofIds,
+        uint16 numOffChainProofs,
+        SubmissionProof[] calldata submissionProofs,
+        uint256 offChainSubmissionMarkers,
+        uint256 duplicateSubmissionIndices
+    ) external onlyWorker {
+        // Expected to fit in a uint16 to match the proof counts.
+        require(proofIds.length <= type(uint16).max, TooManyProofIds());
+
+        VerifierStorage storage verifierStorage = _getVerifierStorage();
+
+        // require there is an outerVerifier contract
+        require(
+            verifierStorage.outerVerifier != address(0),
+            OuterVerifierAddressIsZero()
+        );
+
+        // Track the proof indices to ensure proofs are verified in order.
+        uint64 nextSubmissionIdx = verifierStorage.nextSubmissionIdxToVerify;
+        uint64 verifiedSubmissionHeight = verifierStorage
+            .lastVerifiedSubmissionHeight;
+
+        uint16 submissionProofIdx = 0; // idx into submissionProofs
+        uint16 proofIdIdx = 0; // idx into proofIds
+
+        // Process the off-chain proofIds first
+        for (; proofIdIdx < numOffChainProofs; ++proofIdIdx) {
+            bytes32 proofId = proofIds[proofIdIdx];
+
+            verifierStorage.currentOffChainSubmissionProofIds.push(proofId);
+
+            bool isEndOfSubmission = UpaInternalLib.marksEndOfSubmission(
+                proofIdIdx,
+                offChainSubmissionMarkers
+            );
+
+            if (isEndOfSubmission) {
+                bytes32 submissionId = UpaLib.computeSubmissionId(
+                    verifierStorage.currentOffChainSubmissionProofIds
+                );
+
+                // Only update `verifiedAtBlock` for unverified submissions.
+                // We do not revert the transaction if the submission was
+                // already verified in order to reset the length of
+                // `currentOffChainSubmissionProofIds`. Otherwise, if the
+                // submission was verified over the course of multiple
+                // `verifyAggregatedProof` calls, only the last call would
+                // revert, leaving `currentOffChainSubmissionProofIds` stuck
+                // in an intermediate state.
+                if (verifierStorage.verifiedAtBlock[submissionId] == 0) {
+                    verifierStorage.verifiedAtBlock[submissionId] = block
+                        .number;
+
+                    emit SubmissionVerified(submissionId);
+                }
```

Comment:
@akileshtangella Just flagging this so you are aware in the proof explorer.  We don't necessarily get an event if the submission was already verified.  We still have to compute the submission id and mark that in the DB, even if the corresponding event is not present.

Reply:
@akileshtangella this assumption will break our logic where we check for events to mark submission verified in DB.
Any thoughts?